bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/handler/NukiOpenerHandler.java

```diff
@@ -0,0 +1,278 @@
+    private @Nullable ScheduledFuture<?> reInitJob;
+    private @Nullable String nukiId;
+
+    public NukiOpenerHandler(Thing thing) {
+        super(thing);
+        logger.debug("Instantiating NukiOpenerHandler({})", thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize() for Opener[{}].", getThing().getUID());
+        Configuration config = getConfig();
+        nukiId = (String) config.get(NukiBindingConstants.CONFIG_NUKI_ID);
+        if (nukiId == null) {
+            logger.debug("NukiSmartLockHandler[{}] is not initializable, nukiId setting is unset in the configuration!",
+                    getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "nukiId setting is unset");
+        } else {
+            scheduler.execute(this::initializeHandler);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("dispose() for Smart Lock[{}].", getThing().getUID());
+        stopReInitJob();
+    }
+
+    private void initializeHandler() {
+        logger.debug("initializeHandler() for Opener[{}]", nukiId);
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            initializeHandler(null, null);
+        } else {
+            initializeHandler(bridge.getHandler(), bridge.getStatus());
+        }
+    }
+
+    private void initializeHandler(@Nullable ThingHandler bridgeHandler, @Nullable ThingStatus bridgeStatus) {
+        if (bridgeHandler != null && bridgeStatus != null) {
+            if (bridgeStatus == ThingStatus.ONLINE) {
+                nukiHttpClient = ((NukiBridgeHandler) bridgeHandler).getNukiHttpClient();
+                BridgeLockStateResponse bridgeLockStateResponse = nukiHttpClient.getBridgeOpenerState(nukiId);
+                if (handleResponse(bridgeLockStateResponse, null, null)) {
+                    updateStatus(ThingStatus.ONLINE);
+                    for (Channel channel : thing.getChannels()) {
+                        handleCommand(channel.getUID(), RefreshType.REFRESH);
+                    }
+                    stopReInitJob();
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                stopReInitJob();
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+            stopReInitJob();
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged({}) for Opener[{}].", bridgeStatusInfo, nukiId);
+        scheduler.execute(() -> {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                initializeHandler(null, bridgeStatusInfo.getStatus());
+            } else {
+                initializeHandler(bridge.getHandler(), bridgeStatusInfo.getStatus());
+            }
+        });
```

Comment:
This code is functionally identical.
```suggestion
        scheduler.execute(this::initializeHandler);
```